x-pack/plugins/spaces/server/lib/saved_objects_client/spaces_saved_objects_client.js

```diff
@@ -5,25 +5,17 @@
  */
 
 import { DEFAULT_SPACE_ID } from '../../../common/constants';
-import { isTypeSpaceAware } from './lib/is_type_space_aware';
-import { getSpacesQueryFilters } from './lib/query_filters';
-import uniq from 'lodash';
-import uuid from 'uuid';
 
 export class SpacesSavedObjectsClient {
```

Comment:
We don't really have a mechanism for this, but the `SpacesSavedObjectsClient` has effectively turned into a namespace provider. I'm not recommending changing this now, but maybe this is something we can research in a followup release? It'd be one less Saved Objects Client/Wrapper to maintain...

Reply:
It has... I've started to think of the SavedObjectsClientWrappers more as "interceptors" or "middleware", especially given the priority queue that we're implementing for securing spaces. I think allowing consumers to specify arbitrary "interceptors" with a specified order and anticipating them augmenting the parameters that are being passed through the various interceptors is a relatively generic design pattern that seems to satisfy all of our needs currently.